Handler/Sites.hs

```diff
@@ -97,6 +96,14 @@ siteForm msite = renderBootstrap3 BootstrapBasicForm $ Site
     bfl :: Text -> FieldSettings site
     bfl = bfs
 
+    baseUrlField :: Field Handler Text
+    baseUrlField = flip checkM textField $ \url -> do
```

Comment:
For what it's worth, this is what we do on Rails app (allow a race condition).

No valid data is allowed into the database, and the only way somebody will trigger the race condition is by rapidly resubmitting the form.